app/controllers/jams_controller.rb

```diff
@@ -1,24 +1,43 @@
 # frozen_string_literal: true
 
 class JamsController < ApplicationController
+  before_action :set_room
+
   # POST /rooms/:public_id/jams
   def create
-    room = Room.find_by!(public_id: params[:room_id])
-    jam = room.jams.build(jam_params)
+    jam = @room.jams.build(jam_params)
 
     jam.user = current_user
 
+    # automatically promote uploaded jams of type mix
+    jam.promoted_at = Time.current if jam.mix?
+
     if jam.save
       NotificationService.notify_on_jam_creation(jam, current_user)
       flash[:success] = 'Jam successfully created!'
     else
       flash[:danger] = jam.errors.full_messages.join(', ')
     end
-    redirect_to room_path(room)
+    redirect_to room_path(@room)
+  end
+
+  # POST /rooms/:public_id/jams/:id/promote
+  def promote
+    jam = @room.jams.find(params[:id])
+    if jam.mix? || jam.idea?
```

Comment:
I'd define `.promotable?` on `Jam`

Reply:
even better, have a validation that checks for it so you can just call `if jam.promote` and, since, we use `.update` (without `!`) it'd return false if it fails validations